mongomock/aggregate.py

```diff
@@ -585,6 +586,63 @@ def _handle_unwind_stage(in_collection, unused_database, options):
     return unwound_collection
 
 
+# TODO(pascal): Combine with the equivalent function in collection but check
+# what are the allowed overriding.
+def _combine_projection_spec(filter_list, original_filter, prefix=''):
+    """Re-format a projection fields spec into a nested dictionary.
+
+    e.g: {'a': 1, 'b.c': 1, 'b.d': 1} => {'a': 1, 'b': {'c': 1, 'd': 1}}
```

Comment:
Maybe add that input `filter_list` is the list of keys to keep, not the $project object itself

Reply:
Done.